src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ExplosionsListener.java

```diff
@@ -83,23 +82,23 @@ private void removeResistantBlock(@Nonnull Block block) {
     private void removeResistantBlock(@Nonnull Block block, @Nonnull SlimefunItem slimefunItem) {
         // Fixes #3414 - This check removes the ghost block created by withers.
         if (!(slimefunItem instanceof WitherProof)
-            && !slimefunItem.callItemHandler(BlockBreakHandler.class, handler -> handleExplosion(handler, block))
+            && !slimefunItem.callItemHandler(BlockBreakHandler.class, handler -> handleExplosion(handler, block, slimefunItem))
         ) {
             BlockStorage.clearBlockInfo(block);
             block.setType(Material.AIR);
         }
     }
 
     @ParametersAreNonnullByDefault
-    private void handleExplosion(BlockBreakHandler handler, Block block) {
+    private void handleExplosion(BlockBreakHandler handler, Block block, SlimefunItem sfItem) {
         if (handler.isExplosionAllowed(block)) {
             BlockStorage.clearBlockInfo(block);
             block.setType(Material.AIR);
 
             List<ItemStack> drops = new ArrayList<>();
             handler.onExplode(block, drops);
 
-            for (ItemStack drop : drops) {
+            for (ItemStack drop : sfItem.getDrops()) {
```

Comment:
This will break stuff right? If any current implementation adds items to that list to be dropped this just no longer drops them.

Reply:
In core Slimefun the ``drops`` argument of ``BlockBreakHandler::onExplode`` is never used, that's why I did it this way. But you're right that this doesn't account for custom handlers in addons. Perhaps a better way to implement this is in ``SimpleBlockBreakHandler::onExplode``? Reverting these changes and adding the following code there.
```java
        SlimefunItem sfItem = BlockStorage.check(b);
        if (sfItem != null)
            drops.addAll(sfItem.getDrops());
```
That would also work (already tested) and shouldn't interfere with current and future implementations.